lib/modules/datasource/index.ts

```diff
@@ -415,6 +415,36 @@ export async function getPkgReleases(
             version.matches(constraintValue, releaseConstraint)
         );
       });
+    } else {
+      if (constraintName === 'php') {
```

Comment:
Ideally this hardcoded logic wouldn't be here and this can be supported any time the versioning supports `.subset()` and the range is valid?

Reply:
Ideally, yes. Do you want me to rebase this PR on top of `v35` branch and make it more generic ?

Reply:
Yes, please do. As this feature will sit behind a config option, it should be OK to enable composer and npm functionality in one go

Reply:
@rarkins Done.